src/ast/ast.cpp

```diff
@@ -376,7 +376,7 @@ void AttachPoint::set_index(int index)
 
 std::string Probe::args_typename() const
 {
-  return "struct " + orig_name + "_args";
+  return "struct " + orig_name + attach_points.front()->func + "_args";
```

Comment:
```suggestion
  return "struct " + orig_name + "_" + attach_points.front()->func + "_args";
```
Will be more readable.

On a different note, do we need `orig_name` here? Shouldn't the attach point suffice?

```suggestion
  return "struct " + attach_points.front()->func + "_args";
```

Reply:
> On a different note, do we need orig_name here? Shouldn't the attach point suffice?

I'm not sure if that's unique enough. TBH, I'm a little afraid of how many things this little string might be carrying in terms of importance with type resolution.

Reply:
Yeah, same here, though it sounds like we now always expand probes with `args` so it should always hold that a probe with `args` has a single attach point (i.e. either of those is unique).

Reply:
> like we now always expand probes with args so

Not quite. It's conditional. If the args are of the exact same struct, we don't expand. Here is an example: `tracepoint:xfs:xfs_read_ag*`:
```
$ sudo bpftrace -lv "tracepoint:xfs:xfs_read_ag*"
tracepoint:xfs:xfs_read_agf
    dev_t dev
    xfs_agnumber_t agno
tracepoint:xfs:xfs_read_agi
    dev_t dev
    xfs_agnumber_t agno
```